pkg/secrets/keyvault/client.go

```diff
@@ -183,7 +194,57 @@ func (k *SecretClient) Upsert(ctx context.Context, key secrets.SecretKey, data m
 			SecretAttributes: &secretAttributes,
 		}
 
-		_, err = k.KeyVaultClient.SetSecret(ctx, vaultBaseURL, secretBaseName, secretParams)
+		err = k.setSecret(ctx, secretBaseName, secretParams)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isErrSecretWasSoftDeleted(err error) bool {
+	var azureErr = &azure.RequestError{}
+	if errors.As(err, &azureErr) {
+		if azureErr == nil && azureErr.ServiceError == nil {
```

Comment:
I don't think `azureErr` can be nil at this point. Which is lucky because `azureErr.ServiceError` would panic, right? Should this have been `||`?

Reply:
lol yes this should have been `||`

Reply:
but I'll go ahead and just drop this initial check as you're right, I don't think it can be `nil` here either.

Reply:
(kept the `azureErr.ServiceErrpr == nil` bit, just dropped the `azureErr == nil` bit)